registry/access_control/rbac/auth.py

```diff
@@ -58,20 +58,29 @@ def _get_user_from_token(decoded_token: Mapping) -> User:
         name = decoded_token.get("name", "")
 
         if aad_user_key in decoded_token:
+            # Id Tokens from browser
             username = decoded_token.get(aad_user_key)
             type = UserType.AAD_USER
-        elif aad_app_key in decoded_token.keys:
-            username = decoded_token.get(aad_app_key)
-            name = decoded_token.get("app_displayname", "")
-            type = UserType.AAD_APP
-        elif common_user_key in decoded_token.keys:
+        elif aad_app_key in decoded_token:
+            appid = decoded_token.get(aad_app_key)
+            # Azure CLI User Impersonation token
+            if decoded_token.get("scp") == "user_impersonation":
```

Comment:
can you do == UserType.USER_IMPERSONATION

Reply:
updated